test/e2e/scale/azdisk_scale_test.go

```diff
@@ -0,0 +1,205 @@
+	)
+
+	ginkgo.BeforeEach(func() {
+		cs = f.ClientSet
+		ns = f.Namespace
+		testDriver = driver.InitAzureDiskDriver()
+	})
+
+	ginkgo.It("Scale test scheduling and starting multiple pods with a persistent volume requests. [Windows]", func() {
+		e2eutils.SkipIfUsingInTreeVolumePlugin()
+		e2eutils.SkipIfNotUsingCSIDriverV2()
+
+		pods := []testsuites.PodDetails{}
+		podCount := 1000
+		for itr := 0; itr < podCount; itr++ {
+			volume := testsuites.VolumeDetails{
+				FSType:    "ext3",
+				ClaimSize: "10Gi",
+				VolumeMount: testsuites.VolumeMountDetails{
+					NameGenerate:      "test-volume-" + strconv.Itoa(itr),
+					MountPathGenerate: "/mnt/test-" + strconv.Itoa(itr),
+				},
+			}
+
+			pod := testsuites.PodDetails{
+				Cmd:       e2eutils.ConvertToPowershellorCmdCommandIfNecessary("echo 'hello world' > /mnt/test-1/data && grep 'hello world' /mnt/test-1/data"),
+				Volumes:   e2eutils.NormalizeVolumes([]testsuites.VolumeDetails{volume}, []string{}, isMultiZone),
+				IsWindows: e2eutils.IsWindowsCluster,
+			}
+
+			pods = append(pods, pod)
+
+		}
+		test := testsuites.PodSchedulingWithPVScaleTest{}
+
+		test.CSIDriver = testDriver
+		test.Pods = pods
+		test.PodCount = podCount
+		test.StorageClassParameters = map[string]string{"skuName": "StandardSSD_LRS"}
+
+		test.Run(cs, ns, schedulerName)
+	})
+
+	ginkgo.It("Scale test scheduling and rescheduling multiple pod with a persistent volume request on failover. [Windows]", func() {
+		e2eutils.SkipIfUsingInTreeVolumePlugin()
+		e2eutils.SkipIfNotUsingCSIDriverV2()
+
+		podCount := 1000
+		pods := []testsuites.PodDetails{}
+
+		for itr := 0; itr < podCount; itr++ {
+
+			volume := testsuites.VolumeDetails{
+				FSType:    "ext3",
+				ClaimSize: "10Gi",
+				VolumeMount: testsuites.VolumeMountDetails{
+					NameGenerate:      "test-volume-" + strconv.Itoa(itr),
+					MountPathGenerate: "/mnt/test-" + strconv.Itoa(itr),
+				},
+				MountOptions: []string{
+					"barrier=1",
+					"acl",
+				},
+			}
+
+			pod := testsuites.PodDetails{
+				Cmd:       e2eutils.ConvertToPowershellorCmdCommandIfNecessary("while true; do echo $(date -u) >> /mnt/test-1/data; sleep 3600; done"),
+				Volumes:   e2eutils.NormalizeVolumes([]testsuites.VolumeDetails{volume}, []string{}, isMultiZone),
+				IsWindows: e2eutils.IsWindowsCluster,
+				UseCMD:    false,
+			}
+
+			pods = append(pods, pod)
+		}
+
+		test := testsuites.PodSchedulingOnFailoverScaleTest{}
+		test.CSIDriver = testDriver
+		test.Pods = pods
+		test.PodCount = podCount
+		test.StorageClassParameters = map[string]string{"skuName": "StandardSSD_LRS"}
+
+		test.Run(cs, ns, schedulerName)
+	})
+
+	ginkgo.It("Scale test scheduling and starting multiple pod with multiple persistent volume requests. [Windows]", func() {
+		e2eutils.SkipIfUsingInTreeVolumePlugin()
+		e2eutils.SkipIfNotUsingCSIDriverV2()
+
+		podCount := 350
+		volMountedToPod := 3
+		pods := []testsuites.PodDetails{}
+
+		for i := 1; i <= podCount; i++ {
```

Comment:
We don't need to create 300 copies of the same pod definition. We only need one definition that we later use to create the stateful sets. There is no need to iterate on the number of pods. We only need to iterate on number of volumes only if we want to attach more than one disk per pod.

Reply:
Fixed